cpp/common/src/codingstandards/cpp/rules/deadcode/DeadCode.qll

```diff
@@ -108,17 +105,33 @@ predicate isDeadStmt(Stmt s) {
   exists(TryStmt ts | s = ts and isDeadStmt(ts.getStmt()))
 }
 
-query predicate problems(Stmt s, string message) {
-  not isExcluded(s, getQuery()) and
+/**
+ * Holds if the `Stmt` `s` is dead, i.e. could be executed, but its removal would not meaningfully
+ * affect the program.
+ */
+class DeadStmtInstance extends Stmt {
+  DeadStmtInstance() {
+    isDeadStmt(this) and
+    // Exclude compiler generated statements
+    not this.isCompilerGenerated() and
+    // Exclude code fully generated by macros, because the code may be "live" in other expansions
+    isNotWithinMacroExpansion(this) and
+    // MISRA defines dead code as an "_executed_ statement whose removal would not affect the program
+    // output". We therefore exclude unreachable statements as they are, by definition, not executed.
+    not this.getBasicBlock() = any(UnreachableBasicBlock ubb).getABasicBlock()
+  }
+}
+
+class DeadStmt = HoldsForAllInstances<DeadStmtInstance>::LogicalResultStmt;
```

Comment:
Question, should LogicalResultStmt `extend Stmt` so that the `Stmt`/`StmtInstance` distinction isn't necessary?

Reply:
`LogicalResultElement` can't currently extend `Stmt` (or `Element` as it s now), because it extends the `newtype` that represents the set of all elements that occur at that location, which is not itself a `Stmt` (or `Element`). Instead, it has a one-to-many relationship, exposed by `getAnElementInstance()`.

The alternative would be to arbitrarily pick a specific element that occurs at that location. That would probably work as well, but I think it's more consistent with our intentions to report a logical result and ask the user to get the instances if they want them.